test/unit/database_tables.js

```diff
@@ -194,4 +194,288 @@ describe('DatabaseTables', function() {
             });
         });
     });
+
+    describe('getTables', function () {
+        const scenarios = [
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone'},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34'
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: false,
+                expectedLength: 0
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: null},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34'
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: false,
+                expectedLength: 0
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: undefined},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34'
+                    }
+                ],
+                skipNotUpdatedAtTables: true,
+                skipAnalysisCachedTables: false,
+                expectedLength: 0
+            },
+            {
+                result: [
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tableone', updated_at: Date.now()},
+                    {dbname: 'db1', schema_name: 'public', table_name: 'tabletwo'},
+                    {
+                        dbname: 'db1',
+                        schema_name: 'public',
+                        table_name: 'analysis_b194a8f896_81cc00c1cfbd5c04d3375fc0e0343a34ae979f34',
+                        updated_at: Date.now()
```

Comment:
Ok for the sake of the test, but that situation won't normally happen, right? I mean, it won't have an entry in cdb_tablemetadata unless I cartodbfy the analysis table, which is weird, right?